packages/synapse-react-client/src/components/DataGrid/SynapseGrid.tsx

```diff
@@ -258,9 +206,30 @@ const SynapseGrid = forwardRef<
     }
   }
 
+  const onApplyModelChange = useCallback(
+    (change: ModelChange) => {
+      if (!model) {
+        console.error('Model is not initialized')
+        return
+      }
+
+      applyModelChange(model, change)
+      commit()
+    },
+    [model, commit],
+  )
+
+  const { undoUI, addOperationsToUndoStack } = useGridUndo(
+    model,
+    onApplyModelChange,
+    onUndo,
+  )
```

Comment:
This will let you get rid of the `onUndo` callback. Preserve the `onApplyModelChange` name in `useGridUndo`, but rename the callback variable here.

```suggestion
  const applyModelChangeFromUndo = useCallback(
    (change: ModelChange) => {
      if (!model) {
        console.error('Model is not initialized')
        return
      }
      
      if (change.type === 'DELETE' & gridRef.current) {
        // The user may have set a cell as active that we are removing with an 'undo'. In that case, clear the active state
        gridRef.current.setActiveCell(null)
      }
      
      applyModelChange(model, change)
      commit()
    },
    [model, commit],
  )

  const { undoUI, addOperationsToUndoStack } = useGridUndo(
    model,
    applyModelChangeFromUndo,
  )
```